### PR TITLE
Fixes #6672: Allow plugins to more easily take advantage of resource loo...

### PR DESCRIPTION
**Pull request by ehelms**
I have not added tests yet as I am looking for some direction and opinions before I proceed further. An example of how this changes can be taken advantage of can be seen here - https://github.com/Katello/katello/pull/4454

### CODE REVIEW

File: app/controllers/api/base_controller.rb [hunk 1]
```diff
@@ -247,12 +259,20 @@ def find_nested_object
...
+      messages = []
+
       allowed_nested_id.each do |obj_id|
         if params[obj_id].present?
-          not_found "#{obj_id.humanize} not found by id '#{params[obj_id]}'"
+          messages << "#{obj_id.humanize} not found by id '#{params[obj_id]}'"
         end
       end
+
+      not_found(:message => messages) and return false if !messages.empty?
```
Comment (domcleal) [inline app/controllers/api/base_controller.rb:275]: Slight behaviour and response change here needs reviewing I think.

**Comment by domcleal** [comment]
@isratrade could you review please?  These have been changed a number of times, so I'm a bit worried about anybody beginning to rely on them, that's all...

**Comment by dLobatog** [comment]
@isratrade @ehelms Ping? The code in :develop has changed significantly so I guess a rebase will be complicated unfortunately.

**Comment by tbrisker** [comment]
@ehelms,
Thank you very much for your efforts! 
This PR has been open for a very long time with no activity. 
Please indicate if you plan to continue working on it or not; if not, feel free to close it. 
If you do not respond either way within a couple of weeks, I will assume this PR is abandoned and will go ahead and close it for you.

**Pull request closed without merging**